docs/manual/scala/guide/build/ScalaComponents.md

```diff
@@ -0,0 +1,96 @@
+# Scala Components
+
+The Lagom scaladsl requires building an Application that mixes in all the `Components` that will enable the required features for your code. This means each `Application` will need a specific set of cake slices. Not only that but `Dev` mode and `Prod` mode need adjusting since the environments where the Service is run vary.
+
+Imagine you have a `HelloService` and it's implementation and you are only missing the Appliciation. You could prepare:
+
+@[lagom-application](code/ServiceImplementation.scala)
+
+That is a simple enough Application but it's not complete for running. It extends [`LagomApplication`](api/com/lightbend/lagom/scaladsl/server/LagomApplication.html) so it has all the basic bits and pieces but you need to specify at least a [[Service Locator|ServiceLocator]] for each environment Lagom supports:
+
+@[lagom-loader](code/ServiceImplementation.scala)
+
+Once the `Application` was ready the `ApplicationLoader` is responsible to instantiate and provide the final bits of the cake. See how the `ApplicationLoader` defines two methods: `loadDevMode()` and `load()` so you can mix-in different ServiceLocators.
+
+This is a brief introduction to Lagom Application cake, there's more information in [[Wiring together a Lagom application|ServiceImplementation#Wiring-together-a-Lagom-application]] section.
+
+## Components
+
+This is a list of available `Components` you will need to build your application cake.
+
+[LagomServerComponents](api/com/lightbend/lagom/scaladsl/server/LagomServerComponents.html)
+
+ *  a main `Component` for any Lagom Service and is already mixed into [`LagomApplication`](api/com/lightbend/lagom/scaladsl/server/LagomApplication.html), you won't need to use it explicitly. Includes [LagomServiceClientComponents](api/com/lightbend/lagom/scaladsl/client/LagomServiceClientComponents.html). See [[Wiring together a Lagom application|ServiceImplementation#Wiring-together-a-Lagom-application]].
+
+[LagomServiceClientComponents](api/com/lightbend/lagom/scaladsl/client/LagomServiceClientComponents.html)
+
+ * a main `Component` for any Lagom Service or application consuming Lagom Services. It is already mixed into [`LagomClientApplication`](api/com/lightbend/lagom/scaladsl/client/LagomClientApplication.html), you won't need to use it explicitly. See [[Binding a service client|ServiceClients#Binding-a-service-client]].
```

Comment:
Actually there are a number of use cases for using in explicitly.  See:

https://github.com/lagom/online-auction-scala/blob/master/item-impl/src/main/scala/com/example/auction/item/impl/ItemApplication.scala

Often you may want to break your cake up into isolated layers that can be independently tested.  In the example above, I've created a trait called `ItemComponents` which contains everything necessary to implement `ItemServiceImpl`.  What it doesn't do is subscribe to the bidding service, that's done in `ItemApplication`.  This allows me to use `ItemComponents` to instantiate `ItemServiceImpl` in tests, without needing to worry about mocking the `BiddingService`.